interfaces/builtin/location.go

```diff
@@ -0,0 +1,241 @@
+# Usage: reserved
+
+  # DBus accesses
+  #include <abstractions/dbus-strict>
+  dbus (send)
+     bus=system
+     path=/org/freedesktop/DBus
+     interface=org.freedesktop.DBus
+     member={Request,Release}Name
+     peer=(name=org.freedesktop.DBus),
+
+  dbus (receive, send)
+     bus=system
+     path=/org/freedesktop/DBus
+     interface=org.freedesktop.DBus
+     member=GetConnectionUnixProcessID
+     peer=(label=unconfined),
+
+  dbus (receive, send)
+     bus=system
+     path=/org/freedesktop/DBus
+     interface=org.freedesktop.DBus
+     member=GetConnectionUnixUser
+     peer=(label=unconfined),
+
+  dbus (send)
+    bus=system
+    path=/org/freedesktop/*
+    interface=org.freedesktop.DBus.Properties
+    peer=(label=unconfined),
+
+  # Allow binding the service to the requested connection names
+  dbus (bind)
+      bus=system
+      name="com.ubuntu.location.Service",
+
+  dbus (bind)
+      bus=system
+      name="com.ubuntu.location.Service.Session",
+
+  # Allow traffic to/from our path and interface with any method
+  dbus (receive, send)
+      bus=system
+      path=/com/ubuntu/location/Service{,/**}
+      interface=com.ubuntu.location.Service*,
+
+  dbus (receive, send)
+      bus=system
+      path=/sessions/*
+      interface=com.ubuntu.location.Service.Session
+      member={Start,Stop}PositionUpdates,
+
+  dbus (receive, send)
+      bus=system
+      path=/sessions/*
+      interface=com.ubuntu.location.Service.Session
+      member={Start,Stop}HeadingUpdates,
+
+  dbus (receive, send)
+      bus=system
+      path=/sessions/*
+      interface=com.ubuntu.location.Service.Session
+      member={Start,Stop}VelocityUpdates,
+
+  # Allow traffic to/from org.freedesktop.DBus for location service
+  dbus (receive, send)
+      bus=system
+      path=/
+      interface=org.freedesktop.DBus**,
+  dbus (receive, send)
+      bus=system
+      path=/com/ubuntu/location/Service{,/**}
+      interface=org.freedesktop.DBus**,
+`)
+
+var locationConnectedPlugAppArmor = []byte(`
+# Description: Allow using location service. Reserved because this gives
+#  privileged access to the service.
+# Usage: reserved
+
+#include <abstractions/dbus-strict>
+
+# Allow all access to location service
+dbus (receive, send)
+    bus=system
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+dbus (send)
+    bus=system
+    peer=(name=com.ubuntu.location.Service, label=unconfined),
+
+dbus (send)
+    bus=system
+    peer=(name=com.ubuntu.location.Service.Session, label=unconfined),
+
+dbus (receive)
+    bus=system
+    path=/
+    interface=org.freedesktop.DBus.ObjectManager
+    peer=(label=unconfined),
+
+dbus (receive)
+    bus=system
+    path=/com/ubuntu/location/Service{,/**}
+    interface=org.freedesktop.DBus*
+    peer=(label=unconfined),
+`)
+
+var locationPermanentSlotSecComp = []byte(`
+getsockname
+recvmsg
+sendmsg
+sendto
+`)
+
+var locationConnectedPlugSecComp = []byte(`
+getsockname
+recvmsg
+sendmsg
+sendto
+`)
```

Comment:
@zyga - in some ways I feel like it would be convenient if we could have a function for dbus seccomp slot/plug policy that all these snaps could use, since it is just getting copied around. The downside is it make policy audits quite a bit harder.

That should not block this PR.